onfido/onfido-sdk-ui
# Resolve deprecation & vulnerability warnings & upgrade to Webpack 5

# Problem
There are quite some deprecated packages and vulnerabilities. To fix them we need to upgrade to webpack 5

# Solution
Upgrade webpack to v5 and update other packages to resolve the deprecations and vulnerability alerts

# Changes:
### Added / changes packages: 
- See package.json
### Removed packages: 
- `object-loops` (no safe version available)
- `@types/terser-webpack-plugin` (included in main pkg)
- `@types/mini-css-extract-plugin` (included in main pkg)
- `mochawesome` (not used)
- `postcss-base64` (default in webpack 5)

### Webpack:
- Upgraded to v5
- Add IE11 polyfill for webpack 5
- Migrate `file-loader` with new webpack Asset Modules
- Replaced `webpack-visualizer-plugin [unmaintained]` with `webpack-visualizer-plugin2 [webpack 5 support]` 
- Remove `LoaderOptionsPlugin` which is only meant for migration from webpack `v1` to `v2` ([docs](https://v4.webpack.js.org/plugins/loader-options-plugin/))
- Remove `*.LICENSE.txt` from `dist` (duplicated info, see licenses.json instead)
- Set `fullySpecified` to `false` to support our `castor` packages who aren't fully following `esm` specification
- Use `thread-loader` to speed up `babel-loader`
- Only use `postcss & sass loaders` for non node_modules
- Cleanup `.babelrc`

### Testing & Linting:
- Upgrade `jest`, `eslint` & `mocha` packages to resolve deprecation warnings
- Fix CrossDeviceIntro screen scrolling

### Others
- Add files for bundlewatch
- Support OnfidoAuth in `SdkMount`
- Support `hostedUrl` for browserstack tests

## Bundle:
- Bundle size changed from `250.73KB` ([v6.18.0](https://service.bundlewatch.io/results?d=details%7CrepoOwner%7Confido%7CrepoName%7Confido-sdk-ui%7CrepoCurrentBranch%7Crelease/6.18.0%7CrepoBranchBase%7Cmaster%7CcommitSha%7C7014d7a5cf8f697766c33b120c70e17475b5a5a8%7Cresults%7Cstatus%7Cpass%7CfullResults%7CfilePath%7C./dist/onfido.min.js%7Csize%7CbaseBranchSize%7CmaxSize%7C./dist/onfido.crossDevice.min.js%7C./dist/style.css%7Csummary%7CEverything%2Bis%2Bin%2Bcheck%2B(%252B19.08KB,%2B-22B,%2B%252B5.9%2525)%5E5I3S%7C534Z%7C5J40%7CYT%7CZF%7C10Z%7C1XJV%7C1XFX%7C270G%5E%5E$0%7C$1%7C2%7C3%7C4%7C5%7C6%7C7%7C8%7C9%7CA%5D%7CB%7C$C%7CD%7CE%7C@$F%7CG%7CC%7CD%7CH%7CO%7CI%7CP%7CJ%7CQ%5D%7C$F%7CK%7CC%7CD%7CH%7CR%7CI%7CS%7CJ%7CT%5D%7C$F%7CL%7CC%7CD%7CH%7CU%7CI%7CV%7CJ%7CW%5D%5D%7CM%7CN%5D%5D)) to `265.45KB` ([PR](https://service.bundlewatch.io/results?d=details%7CrepoOwner%7Confido%7CrepoName%7Confido-sdk-ui%7CrepoCurrentBranch%7Cwebpack-upgrade-CX7288%7CrepoBranchBase%7Cdevelopment%7CcommitSha%7C52112330035f907e363ac1c286689788db029711%7Cresults%7Cstatus%7Cpass%7CfullResults%7CfilePath%7C./dist/onfido.min.js%7Csize%7CbaseBranchSize%7CmaxSize%7C./dist/onfido.crossDevice.min.js%7C./dist/onfido.vendors~crossDevice.min.js%7C./dist/style.css%7C./dist/onfido.crossDevice.css%7Csummary%7CEverything%2Bis%2Bin%2Bcheck%2B(%252B419.81KB,%2B-0B)%5E5TQM%7C0%7C5XC0%7CZV%7C0%7C10Z%7C1EBY%7C0%7C1JB4%7C1XLE%7C0%7C270G%7C11G%7C0%7C1KW%5E%5E$0%7C$1%7C2%7C3%7C4%7C5%7C6%7C7%7C8%7C9%7CA%5D%7CB%7C$C%7CD%7CE%7C@$F%7CG%7CC%7CD%7CH%7CQ%7CI%7CR%7CJ%7CS%5D%7C$F%7CK%7CC%7CD%7CH%7CT%7CI%7CU%7CJ%7CV%5D%7C$F%7CL%7CC%7CD%7CH%7CW%7CI%7CX%7CJ%7CY%5D%7C$F%7CM%7CC%7CD%7CH%7CZ%7CI%7C10%7CJ%7C11%5D%7C$F%7CN%7CC%7CD%7CH%7C12%7CI%7C13%7CJ%7C14%5D%5D%7CO%7CP%5D%5D)) due to polyfills

# How to test this PR:
- Checkout this branch on your local machine
- Run `npm i` 
- Run `npm run build`
- Run `npm run build:auth`
- Run `npm run build:test`
- Run `npm run dev`
- Run `npm run dev:auth`
- Check the app in the browser to check if it works in general

> In case you get into any trouble with packages, run `rm -rf node_modules` & `npm i`



# Checklist
_put `n/a` if item is not relevant to PR changes_

- [x] Has the CHANGELOG been updated?
- [ ] Has the README been updated?
- [ ] Has the CONTRIBUTING doc been updated?
- [ ] Has the RELEASE_GUIDELINES been updated?
- [ ] Has the TESTING_GUIDELINES been updated?
- [ ] Has the MIGRATION doc been updated for any MAJOR breaking changes?
- [ ] Has the MIGRATION doc been updated for any MINOR breaking changes, including any translation strings or keys changes?
- [ ] Have any new automated tests been implemented or the existing ones changed?
- [ ] Have any new manual tests been written down or the existing ones changed?
- [ ] Have any new strings been translated or the existing ones changed?
